docs/guidance/rayservice.md

```diff
@@ -0,0 +1,202 @@
+## Ray Services (alpha)
+
+> Note: This is the alpha version of Ray Services. There will be ongoing improvements for Ray Services in the future releases.
+
+### Prerequisite
+
+* Ray 2.0 is required.
+
+### What is a RayService?
+
+The RayService is a new custom resource (CR) supported by KubeRay in v0.3.0.
+
+A RayService manages 2 things:
+* RayCluster: Manages resources in kubernetes cluster.
+* Ray Serve Deployment Graph: Manages users' serve deployment graph.
+
+### What does the RayService provide?
+
+* Kubernetes-native support for Ray cluster and Ray Serve deployment graphs. You can use a kubernetes config to define a ray cluster and its ray serve deployment graphs. Then you can use `kubectl` to create the cluster and its graphs.
+* In-place update for ray serve deployment graph. Users can update the ray serve deployment graph config in the RayService CR config and use `kubectl apply` to update the serve deployment graph.
+* Zero downtime upgrade for ray cluster. Users can update the ray cluster config in the RayService CR config and use `kubectl apply` to update the ray cluster. RayService will temporarily create a pending ray cluster, wait for the pending ray cluster ready, and then switch traffics to the new ray cluster, terminate the old cluster. 
+* Services HA. RayService will monitor the ray cluster and serve deployments health status. If RayService detects any unhealthy status lasting for a certain time, RayService will try to create a new ray cluster, and switch traffic to the new cluster when it is ready.
+
+### Deploy the Operator
+
+`$ kubectl apply -k "github.com/ray-project/kuberay/ray-operator/config/default"`
+
+Check that the controller is running.
+
+```shell
+$ kubectl get deployments -n ray-system
+NAME           READY   UP-TO-DATE   AVAILABLE   AGE
+ray-operator   1/1     1            1           40s
+
+$ kubectl get pods -n ray-system
+NAME                            READY   STATUS    RESTARTS   AGE
+ray-operator-75dbbf8587-5lrvn   1/1     Running   0          31s
+```
+
+### Run an example cluster
+
+There is one example config file to deploy RaySerive included here:
+[ray_v1alpha1_rayservice.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray_v1alpha1_rayservice.yaml)
+
+```shell
+# Create a ray service and deploy fruit deployment graph.
+$ kubectl apply -f config/samples/ray_v1alpha1_rayservice.yaml
+```
+
+```shell
+# List running RayServices.
+$ kubectl get rayservice
+NAME                AGE
+rayservice-sample   7s
+```
+
+```shell
+# The created RayService should include a head pod, a worker pod, and four services.
+$ kubectl get pods
+NAME                                                      READY   STATUS    RESTARTS   AGE
+ervice-sample-raycluster-qd2vl-worker-small-group-bxpp6   1/1     Running   0          24m
```

Comment:
The fact that this name is getting cut off for a basic example is a symptom of all of the layers involved here
service -> cluster -> worker group -> pod

I wonder if we could shorten some names to avoid this...

Reply:
Actually there are some user defined names. So it is hard to totally avoid. But we should be able to reduce the chances.